docs: Update documentation for Arbitrum Sepolia, Stylus CLI, and content clarity

omarespejel opened the pull request:
1. Added detailed instructions for adding the Arbitrum Sepolia network to the MetaMask wallet, ensuring users have clear guidance for network setup.
2. Updated the output details for the Stylus CLI to reflect the latest functionalities and expected results, enhancing user understanding.
3. Revised various sections across the documentation to improve clarity and readability, making the content more accessible to users.

omarespejel commented:
FYI, i already signed the Contributor License Agreement.

eljobe commented on arbitrum-docs/stylus/stylus-quickstart.md:
```diff
@@ -11,89 +11,150 @@ import PublicPreviewBannerPartial from './partials/_stylus-public-preview-banner
-For additional sources of testnet ETH, please use a faucet on Arbitrum Sepolia or Ethereum Sepolia:
+1. Visit a faucet such as [Bware Labs' Arbitrum Stylus Testnet faucet](https://bwarelabs.com/faucets/arbitrum-stylus-testnet).
+2. Enter your wallet address.
+3. Click `Claim` to receive your testnet ETH. Some faucets may offer additional steps to get more tokens.
```
I would change "offer" to "require".

chrisco512 commented on arbitrum-docs/stylus/stylus-quickstart.md:
````diff
@@ -103,21 +164,47 @@ Start a new Stylus project with
+```
 
-To check whether or not your program will successfully **deploy and activate** onchain, use the `cargo stylus check` subcommand:
+This creates a project with just the essential Stylus entrypoint locally.
````
Remove as well. See above.

chrisco512 commented on arbitrum-docs/stylus/stylus-quickstart.md:
```diff
@@ -103,21 +164,47 @@ Start a new Stylus project with
+
+### Checking Program Validity
+
+Before deploying your program onchain, ensure its it works:
```
Remove these changes

chrisco512 commented on arbitrum-docs/stylus/stylus-quickstart.md:
````diff
@@ -103,21 +164,47 @@ Start a new Stylus project with
 ```
 
-This command will attempt to verify that your program can be deployed and activated onchain without requiring a transaction by specifying a JSON-RPC endpoint. See `cargo stylus check --help` for more options.
+This command verifies that your program can be deployed and activated onchain without a transaction, by connecting to a specified JSON-RPC endpoint. The defalt endpoint is the Arbitrum Stylus testnet: `https://stylus-testnet.arbitrum.io/rpc`.
````
`default` is misspelled here. Let's keep "See `cargo stylus check --help` for more options." inlined here and skip the callout on lines 189 through 193 below.

chrisco512 commented on arbitrum-docs/stylus/stylus-quickstart.md:
````diff
@@ -130,65 +217,81 @@ Caused by:
+### Deploying Your Program
 
-```
+Once validated, you can deploy your program using the `cargo stylus deploy` subcommand. Start by estimating the gas required for deployment. `--private-key-path` is a text file containing your private key. You could also use the `--private-key` flag to input your private key directly but this is not recommended for security reasons. 
````
Let's remove the `--private-key` flag sentence here. We have no specific recommendation wrt security on using an inline key vs a key path; and it distracts from single prescribed way for basic example.

chrisco512 commented on arbitrum-docs/stylus/stylus-quickstart.md:
````diff
@@ -130,65 +217,81 @@ Caused by:
 ```
 
-Next, attempt an actual deployment. Two transactions will be sent onchain.
+To proceed with the deployment, two transactions will be sent onchain:
````
Remove this change

chrisco512 commented on arbitrum-docs/stylus/stylus-quickstart.md:
```diff
@@ -130,65 +217,81 @@ Caused by:
-More options exist for sending and outputting your transaction data. See `cargo stylus deploy --help` for more details.
 
-## Deploying non-Rust WASM projects
+## Deploying Non-Rust WASM Projects
```
Other than this line I'd remove the changes in this section. It was meant as a short aside, and it probably should be moved to another part of the docs altogether since it's focused on WASM and this page is meant as a Rust quick start guide. We'll revisit closer to mainnet.

chrisco512 commented on arbitrum-docs/stylus/stylus-quickstart.md:
```diff
@@ -198,11 +301,32 @@ For example:
+This feature supports seamless integration of Rust-based smart contracts with other Ethereum tools and frameworks, which often rely on Solidity ABIs to interact with various contracts.
 
-can be saved as `add.wat` and used as `cargo stylus check --wasm-file-path=add.wat` or `cargo stylus deploy --priv-key-path=<YOUR PRIV KEY FILE PATH> --wasm-file-path=add.wat`
+# Conclusion
```
Remove this section. The quick start guide does not teach you to write smart contracts, just to deploy them, and the conclusion section is superfluous.

omarespejel commented on arbitrum-docs/stylus/stylus-quickstart.md:
```diff
@@ -11,89 +11,150 @@ import PublicPreviewBannerPartial from './partials/_stylus-public-preview-banner
-2. Enter your wallet address into the text field.
-3. Click `Claim` and optionally follow the second step to receive extra testnet tokens.
-4. You should now have Sepolia ETH on the Stylus testnet.
+To engage with the Stylus testnet, you will first need to acquire Sepolia ETH and then bridge it over:
```
I agree, I've updated the text accordingly. Indeed, I couldn't use the Bware faucet and obtained my ETH through Alchemy. It turned out to be more challenging than expected to get the test ETH

omarespejel commented on arbitrum-docs/stylus/assets/cargo-stylus.png:
@chrisco512, does this mean I should restore the deleted images? I can do that, imo, I felt they were redundant and didn’t add much value to the content. Let me know how you’d like to proceed!

omarespejel commented:
Thank you, @eljobe and @chrisco512, for your valuable comments! I agree with your suggestions and will incorporate this style and advice into my future contributions to the docs

I've addressed your comments. please let me know if there's anything else you need. Thanks again!

omarespejel commented:
@symbolpunk: thanks for the commits, i'll apply this writing style in a next contribution

omarespejel commented:
@symbolpunk:

* Thanks for the commits. I'll apply this writing style in my next contribution.
* I've already signed the Contributor License Agreement for Arbitrum Nitro twice, using my GitHub username `omarespejel`, but the bot isn't detecting it. I can send directly to an email if you provide one.

omarespejel commented:
Hi @symbolpunk:

Kind reminder on this point:` I've already signed the Contributor License Agreement for Arbitrum Nitro twice, using my GitHub username omarespejel, but the bot isn't detecting it. I can send directly to an email if you provide one.
`

omarespejel commented:
Hey @eljobe @chrisco512 @symbolpunk, 
* I updated the quick start to reflect stylus V2. 
* I've already signed the Contributor License Agreement for Arbitrum Nitro twice, using my GitHub username omarespejel, but the bot isn't detecting it. I can send directly to an email if you provide one.

hkalodner commented:
@cla-bot check

hkalodner commented:
Sorry about the delay on the CLA!

omarespejel commented:
hey @hkalodner! It seems like the PR is still being flagged as blocked due to CLA
